src/Scene/NodeProcess.js

```diff
@@ -290,23 +274,33 @@ function updateNodeImagery(quadtree, node, layersConfig, force) {
                     // and stop retrying after X attempts.
                 }
 
+                node.layerUpdateState[layer.id].success();
+
                 return result;
+            },
+            function(err) {
+                if (err instanceof CancelledCommandException) {
+                    node.layerUpdateState[layer.id].success();
+                } else {
+                    node.layerUpdateState[layer.id].failure(Date.now());
+                    scene.notifyChange(node.layerUpdateState[layer.id].waitDurationUntilNextTry(), false);
```

Comment:
Changing from `notifyChange` function to an event belongs to another PR (if we decide to do the change).